Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -959,30 +975,85 @@ double TimeSeriesProperty<TYPE>::averageValueInFilter(
   // 'Normalise' by the total time
   return numerator / totalTime;
 }
-/** Calculates the time-weighted average of a property.
- *  @return The time-weighted average value of the log.
+
+/** Function specialization for TimeSeriesProperty<std::string>
+ *  @throws Kernel::Exception::NotImplementedError always
  */
+template <>
+double TimeSeriesProperty<std::string>::averageValueInFilter(
+    const TimeSplitterType &) const {
+  throw Exception::NotImplementedError("TimeSeriesProperty::"
+                                       "averageValueInFilter is not "
+                                       "implemented for string properties");
+}
+
 template <typename TYPE>
-double TimeSeriesProperty<TYPE>::timeAverageValue() const {
-  double retVal = 0.0;
+std::pair<double, double>
+TimeSeriesProperty<TYPE>::timeAverageValueAndStdDev() const {
+  std::pair<double, double> retVal{0., 0.}; // mean and stddev
   try {
     const auto &filter = getSplittingIntervals();
-    retVal = this->averageValueInFilter(filter);
+    retVal = this->averageAndStdDevInFilter(filter);
   } catch (std::exception &) {
-    // just return nan
-    retVal = std::numeric_limits<double>::quiet_NaN();
+    retVal.first = std::numeric_limits<double>::quiet_NaN();
+    retVal.second = std::numeric_limits<double>::quiet_NaN();
   }
   return retVal;
 }
 
+template <typename TYPE>
+std::pair<double, double> TimeSeriesProperty<TYPE>::averageAndStdDevInFilter(
```

Comment:
Have you considered using `Boost.Accumulators`?
see Framework/Kernel/src/Statistics.cpp, Line 195

(bonus points, make a specialization in the above file.